Tests/BuildTests/BuildPlanTests.swift

```diff
@@ -3770,17 +3787,17 @@ final class BuildPlanTests: XCTestCase {
 
         // Compile C Target
         let cLibCompileArguments = try result.target(for: "cLib").clangTarget().basicArguments(isCXX: false)
-        let cLibCompileArgumentsPattern: [StringPattern] = ["-I", "\(sdkIncludeSearchPath)"]
+        let cLibCompileArgumentsPattern: [StringPattern] = ["-I", "\(AbsolutePath(sdkIncludeSearchPath)._nativePathString(escaped: false))"]
         XCTAssertMatch(cLibCompileArguments, cLibCompileArgumentsPattern)
 
         // Compile Swift Target
         let exeCompileArguments = try result.target(for: "exe").swiftTarget().compileArguments()
-        let exeCompileArgumentsPattern: [StringPattern] = ["-I", "\(sdkIncludeSearchPath)"]
+        let exeCompileArgumentsPattern: [StringPattern] = ["-I", "\(AbsolutePath(sdkIncludeSearchPath)._nativePathString(escaped: false))"]
         XCTAssertMatch(exeCompileArguments, exeCompileArgumentsPattern)
 
         // Link Product
         let exeLinkArguments = try result.buildProduct(for: "exe").linkArguments()
-        let exeLinkArgumentsPattern: [StringPattern] = ["-L", "\(sdkIncludeSearchPath)"]
+        let exeLinkArgumentsPattern: [StringPattern] = ["-L", "\(AbsolutePath(sdkIncludeSearchPath)._nativePathString(escaped: false))"]
```

Comment:
@tomerd WDYT of adjusting `CustomStringConvertible` and `description` of `AbsolutePath` to use `_nativePathString` under the hood so that we don't have to spell it out explicitly everywhere?

Reply:
@MaxDesiatov - and then add an `escaped` member to escape it?  Or would we do that via a standalone function or would that be an member function?

Reply:
@compnerd - @MaxDesiatov @neonichu @abertelrud and myself discussed this some more and believe the solution should be to change `AbsolutePath::pathString` to return the native path string unescaped then escape where necessary. @MaxDesiatov will follow up with how implement and test this, so that we can get rid of `_nativePathString` and make that behavior the default string representation

Reply:
Partially addressed in https://github.com/apple/swift-package-manager/pull/6730. If you think that's a correct approach I'll expand it to clean up the remaining uses of `_nativePathString` in our codebase.